package.json

```diff
@@ -134,6 +134,11 @@
             "maxNumberOfProblems": 100,
             "trace.server": "off"
           }
+        },
+        "terraform.rootModules": {
+          "scope": "resource",
+          "type": "array",
+          "description": "Per-workspace list of module directories for the language server to read"
```

Comment:
should this have nested proprties, in case we need to support that? not sure if JSON schema can properly convey a string or a property, so to avoid breaking people down the road, maybe we need to go for the nested properties now?

Reply:
By nested you mean the bit under "Optionally we may also want to support per root module expanded configuration." ? I have some concerns about managing this, one of them being that there are eight different layers of configuration scopes that all get rolled up together. The docs note: "Only object value types are merged and all other value types are overridden."

Reply:
The more I think about it we should just deal with that when we get to it and just go with array of strings for now. JSON schema does support `oneOf`, whether VS Code does though I am unsure, and I'd imagine it definitely doesn't support it in the UI.

Reply:
array of strings is currently what the server expects:
https://github.com/hashicorp/terraform-ls/pull/198/files#diff-98ffa6f0726a4bff51793b717c302e0dR13

Changing the type and supporting different types for the same option will probably be difficult, but we can (and probably should, to avoid confusion) deprecate one and introduce another if we need to.